### PR TITLE
Delete rolled back L1 sequence on sequence rollback event

### PR DESCRIPTION
**Pull request by cffls**
Once approved, this PR is safe to be merged because it will only be effective when a sequence rollback event happens on L1. 

Tested on a kurtosis network where a rollback event was manually triggered.

### CODE REVIEW

File: zk/contracts/l1_contracts.go [hunk 1]
```diff
@@ -16,4 +16,5 @@ var (
 	AddNewRollupTypeTopic          = common.HexToHash("0xa2970448b3bd66ba7e524e7b2a5b9cf94fa29e32488fb942afdfe70dd4b77b52")
 	CreateNewRollupTopic           = common.HexToHash("0x194c983456df6701c6a50830b90fe80e72b823411d0d524970c9590dc277a641")
 	UpdateRollupTopic              = common.HexToHash("0xf585e04c05d396901170247783d3e5f0ee9c1df23072985b50af089f5e48b19d")
+	RollbackSequenceTopic          = common.HexToHash("0x1125aaf62d132d8e2d02005114f8fc360ff204c3105e4f1a700a1340dc55d5b1")
```
Comment (Stefan-Ethernal) [inline zk/contracts/l1_contracts.go]: I'd consider renaming it to `RollbackBatchesTopic` (https://github.com/0xPolygonHermez/zkevm-contracts/blob/a9b4f742f66bd4f3bcd98a3a188422480ffe0d4e/contracts/v2/PolygonRollupManager.sol#L328-L332), in order to be consistent with the smart contracts. WDYT?

File: zk/stages/stage_l1syncer.go [hunk 2]
```diff
@@ -214,6 +221,7 @@ var (
 	logSequence         BatchLogType = 1
 	logVerify           BatchLogType = 2
 	logL1InfoTreeUpdate BatchLogType = 4
+	logRollbackSequence BatchLogType = 5
```
Comment (Stefan-Ethernal) [inline zk/stages/stage_l1syncer.go]: WDYT?
```suggestion
	logRollbackBatches BatchLogType = 5
```

File: zk/hermez_db/db.go [hunk 3]
```diff
@@ -536,6 +536,26 @@ func (db *HermezDb) WriteSequence(l1BlockNo, batchNo uint64, l1TxHash, stateRoot
 	return db.tx.Put(L1SEQUENCES, ConcatKey(l1BlockNo, batchNo), val)
 }
 
+func (db *HermezDb) RollbackSequence(batchNo uint64) error {
```
Comment (Stefan-Ethernal) [inline zk/hermez_db/db.go]: Could you add the comment?

**Review by Stefan-Ethernal — DISMISSED** [review]
Other than the minor comments, LGTM!

File: zk/contracts/l1_contracts.go (hunk 1 above)
Comment (cffls) [reply zk/contracts/l1_contracts.go]: Good idea, will change.

File: zk/hermez_db/db.go (hunk 3 above)
Comment (cffls) [reply zk/hermez_db/db.go]: Added a comment

**Review by hexoscott — APPROVED** [review]
Looks good to me

**Pull request merged**